File: Java/Temperatures.java
package Java;
import java.util.*;
import java.math.*;
class Solution {
    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
        
        int N = in.nextInt();
        if (N == 0)
            System.out.println("0");
        else {
            int T;
            int min = Integer.MAX_VALUE;
            for (int i = 0; i < N; i++) {
                T = in.nextInt();
                if (Math.abs(T) < Math.abs(min) || (T == -min && T > 0))
                    min = T;
            }
            System.out.println(min);
        }
        in.close();
    }
}
    

File: Python/Facile/Temperatures.py
N = int(input())
if N == 0:
    print("0")
else:
    min = None
    for T in map(int, input().split()):
        if (min == None) or (abs(T) < abs(min)) or ((T == -min) and (T > 0)):
            min = T
    print(min)